### PR TITLE
[Enhancement] Cisco Aironet add Hashlookup for event.reason

### PR DESCRIPTION
**Pull request by WildDogOne**
## What does this PR do?
This PR adds a hashlookup table for event.reasons
The reasons are gathered from the [Cisco Documentation](https://www.cisco.com/c/en/us/support/wireless/wireless-lan-controller-software/products-system-message-guides-list.html)

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/cisco_aironet/data_stream/log/sample_event.json [hunk 1]
```diff
@@ -24,15 +24,16 @@
...
         "snapshot": false,
-        "version": "8.3.3"
+        "version": "8.4.1"
     },
     "event": {
         "action": "ENTRY_DELETED",
         "agent_id_status": "verified",
         "dataset": "cisco_aironet.log",
-        "ingested": "2022-09-09T08:30:39Z",
+        "description": "Device Smart-Licensing status is out of compliance.",
```
Comment (efd6) [inline packages/cisco_aironet/data_stream/log/sample_event.json:35]: This doesn't look right. I think you need to run `elastic-package test system -g` (and then `elastic-package build` to carry this to the docs).

File: packages/cisco_aironet/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,10 @@
 # newer versions go on top
+
```
Comment (efd6) [inline packages/cisco_aironet/changelog.yml:2]:
```suggestion
```

**Review by efd6 — COMMENTED** [review]
Do you have an idea of what an expected number of lookup table entries are likely to be wanted in the end?

**Comment by WildDogOne** [comment]
> Do you have an idea of what an expected number of lookup table entries are likely to be wanted in the end?

Sadly no idea, for now I hope it won't be much more to add

File: packages/cisco_aironet/data_stream/log/sample_event.json (hunk 1 above)
Comment (WildDogOne) [reply packages/cisco_aironet/data_stream/log/sample_event.json]: no idea why this didn't build correctly, should now be fixed

**Comment by efd6** [comment]
I'm wondering how useful these human-directed phrases will be. We don't have a similar external documentation enrichment feature in other integrations; we normally use the log line itself to obtain the data. I think it certainly makes it friendlier to the user, but I'm concerned about the potential for this list to grow and become a processing burden. This is not to say that it should not be added, but it would be good to get an idea of how many would be a real world expectation of addition and how valuable the addition to the docs would be.

**Comment by WildDogOne** [comment]
@efd6 I will check with our network engineers if this feature helps them at all.
If they believe it to be useless, I will close this pull request.

one question I do have though, would using an enrichment processor make this more efficient?
we use quite a lot of enrichments and on huge tables, with so far not much impact

another idea would be, I could add a flag for enabling and disabling the usage of the hashlookup in the UI

**Comment by efd6** [comment]
Sorry for the slow response. I think an enrichment processor would work well; this would then be up to the user to add the additional index (maybe add to the documentation the location of the mapping file) and include the enrichment processor in the processors configuration.

**Comment by WildDogOne** [comment]
I think the necessity for this has dropped for us
We are using an enrichment set.
I can if someone is interested, add a little tutorial into the readme on how one could set that up
any takers?

**Pull request closed without merging**